src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -1,22 +1,98 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Abstractions;
 using Raven.Server.Routing;
 using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
 
 namespace Raven.Server.Documents.Handlers
 {
     public class DocumentReplicationRequestHandler : DatabaseRequestHandler
     {
-        [RavenAction("/databases/*/documentReplication", "GET", "/databases/{databaseName:string}/documentReplication")]
+        //an endpoint to establish replication websocket
+        [RavenAction("/databases/*/documentReplication", "GET",
+            @"/databases/{databaseName:string}/documentReplication?
+                srcDbId={databaseUniqueId:string}
+                &srcUrl={url:string}
+                &srcDbName={databaseName:string}
+                &lastSentEtag={etag:long}")]
         public async Task DocumentReplicationConnection()
         {
-            DocumentsOperationContext context;
-            
-            using (var webSocket = await HttpContext.WebSockets.AcceptWebSocketAsync())
-            using (ContextPool.AllocateOperationContext(out context))
+            var dbId = Guid.Parse(GetQueryStringValueAndAssertIfSingleAndNotEmpty("srcDbId")[0]);
+            var srcUrl = GetQueryStringValueAndAssertIfSingleAndNotEmpty("srcUrl")[0];
+            var srcDbName = GetQueryStringValueAndAssertIfSingleAndNotEmpty("srcDbName")[0];
+            long lastSentEtag;
+            if (!long.TryParse(GetQueryStringValueAndAssertIfSingleAndNotEmpty("lastSentEtag")[0], out lastSentEtag))
```

Comment:
why are we sending the lastSendEtag  ? The request is going to live for a long time, and this is going to be out of date immeidately.